docs/index.rst

```diff
@@ -3,8 +3,128 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to uarray's documentation!
-==================================
+Welcome to ``uarray``'s documentation
+=====================================
+
+.. warning::
+    ``uarray`` is a developer tool, it is not meant to be used directly by end-users.
+
+.. warning::
+    This document is meant to elicit discussion from the broader community and to help
+    drive the direction that ``uarray`` goes towards. Examples proviced here may not be
+    immediately stable.
+
+``uarray`` is a backend/dispatch mechanism with a focus on array computing and the
+needs of the wider array community. It is meant to address the shortcomings of `NEP-18
+<http://www.numpy.org/neps/nep-0018-array-function-protocol.html>`_ and `NEP-13
+<https://www.numpy.org/neps/nep-0013-ufunc-overrides.html>`_; while also being a
+comprehensive solution to `NEP-22 <http://www.numpy.org/neps/nep-0022-ndarray-duck-typing-overview.html>`_,
+while still holding nothing in ``uarray`` itself that's specific to array computing
+or the NumPy API.
+
+Design Philosophies
+-------------------
+
+The following section discusses the design philosophies of ``uarray``, and the
+reasoning behind some of these philosophies.
+
+Modularity
+^^^^^^^^^^
+
+``uarray`` (and its sister modules ``unumpy`` and others to come) were designed
+from the ground-up to be modular. This is part of why ``uarray`` itself holds
+the core backend and dispatch machinery, and ``unumpy`` holds the actual
+multimethods. Also, ``unumpy`` can be developed completely separately to
+``uarray``, although the ideal place to have it would be NumPy itself.
+
+However, the benefit to having it separate is that it could span multiple
+NumPy versions, even before NEP-18 (or even NEP-13) was available.
```

Comment:
Also, it can have a faster release cycle which may be useful in the first 1-2 years.